internal/runners/activate/activation.go

```diff
@@ -90,6 +104,13 @@ func activate(proj *project.Project, out output.Outputer, subs subshell.SubShell
 		return false, locale.WrapError(fail, "error_could_not_activate_venv", "Could not activate project. If this is a private project ensure that you are authenticated.")
 	}
 
+	if setDefault {
```

Comment:
The `venv.Activate()` method here fails in case we already have an activated project.   So, I cannot go this route when I have an activated project and make it the default project.  But yes, you're right and this can be simplified.

Reply:
I refactored this quite a bit, and I believe that it is clearer now.